src/core/internal/camelCase.ts

```diff
@@ -1,7 +1,17 @@
 function camelCase (kebobCase: string): string {
-  return kebobCase.replace(/-([a-z])/g, (kk) => {
-    return kk[1].toUpperCase()
-  })
+  // make any alphabets that follows '-' an uppercase character, and remove the corresponding hyphen
+  let cameledParam = kebobCase.replace(/-([a-z])/g, (kk) => {
+    return kk[1].toUpperCase();
+  });
+
+  // remove '-' and '.' from the string
+  const outParam = cameledParam.replace(/-|\./g, '')
+
+  // check if resulting parameter name is alpha-numeric and starts with an alphabet
+  if(!outParam.match(/^([a-z])+([0-9a-z]+)$/i)) {
+    console.error(`Resulting parameter name is not alpha-numeric: ${outParam}`)
```

Comment:
Instead of logging whether the parameter name is not valid, how about just stripping all non-valid chars to make outParam valid? I'd imagine most uses of this function will only have alphanumeric chars involved.

Reply:
Ok, makes sense. I have changed the code to just replace all non-alphanumeric characters. I added to check for resulting empty strings.